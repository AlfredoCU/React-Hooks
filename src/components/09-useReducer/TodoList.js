import React, { memo } from "react";
import PropTypes from "prop-types";
import TodoListItem from "./TodoListItem";

const TodoList = memo(({ todos, handleDelete, handleToggle }) => {
  console.log("TodoList");
  return (
    <section className="col-7">
      <ul className="list-group text-info">
        {todos &&
          todos.map((todo, key) => (
            <TodoListItem
              key={todo.id}
              todo={todo}
              index={key}
              handleDelete={handleDelete}
              handleToggle={handleToggle}
            />
          ))}
      </ul>
    </section>
  );
});

TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleToggle: PropTypes.func.isRequired,
};

export default TodoList;
