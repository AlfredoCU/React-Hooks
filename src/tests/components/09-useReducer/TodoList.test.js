import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { testTodos } from "../../fixtures/testTodos";
import TodoList from "../../../components/09-useReducer/TodoList";

describe("Pruebas en el componente TodoList", () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();

  const wrapper = shallow(
    <TodoList
      todos={testTodos}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );

  test("Foto del componente.", () => {
    expect(wrapper).toMatchSnapshot();
  });

  //! Para que funcine se debe de quitar el memo del componente.
  test("Debe de tener dos <TodoListItem/>", () => {
    //! Para que funcione esto, no debe de estar el metodo memo englobando todo un componente.
    //! Ya que TodoListItem no existiria y en su lugar mostaria un memo().
    // expect(wrapper.find("TodoListItem").length).toBe(testTodos.length);
    // expect(wrapper.find("TodoListItem").at(0).prop("handleDelete")).toEqual(
    //   expect.any(Function)
    // );
    // console.log(wrapper.find("TodoListItem").length);
    // console.log(wrapper.find("TodoListItem").at(0).props());
  });
});
