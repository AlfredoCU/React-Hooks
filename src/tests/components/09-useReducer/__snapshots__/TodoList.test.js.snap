// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente TodoList Foto del componente. 1`] = `
<section
  className="col-7"
>
  <ul
    className="list-group text-info"
  >
    <Memo()
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={0}
      key="1"
      todo={
        Object {
          "desc": "Aprender React",
          "done": false,
          "id": 1,
        }
      }
    />
    <Memo()
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={1}
      key="2"
      todo={
        Object {
          "desc": "Aprender Vue",
          "done": false,
          "id": 2,
        }
      }
    />
  </ul>
</section>
`;
